src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,416 @@
+
+# Change of basis between two matrices
+
+###############################################################################################################
+
+# modifies A by eliminating all hyperbolic lines and turning A into a diagonal matrix
+# return the matrix Z such that Z*A*transpose(Z) is diagonal
+function _elim_hyp_lines(A::MatElem{T}) where T <: FinFieldElem
+   F = base_ring(A)
+   n = nrows(A)
+   b = matrix(F,2,2,[1,1,1,-1])  # change of basis from matrix([0,1,1,0]) to matrix([2,0,0,-2])
+   Z = identity_matrix(F,n)
+
+   i = 1
+   while i <= n
+      if A[i,i]==0
+         A[i,i]=2
+         A[i,i+1]=0
+         A[i+1,i]=0
+         A[i+1,i+1]=-2
+         insert_block!(Z,b,i,i)
+         i+=2
+      else
+         i+=1
+      end
+   end
+
+   return Z
+end
+
+
+# return true, D such that D*B2*conjugatetranspose(D)=B1
+# return false, nothing if D does not exist
+# TODO: orthogonal only in odd char, at the moment
+function _change_basis_forms(B1::MatElem{T}, B2::MatElem{T}, _type::Symbol)  where T <: FinFieldElem
+
+   if _type==:alternating || _type==:hermitian
+      D1 = _to_standard_form(B1,_type)
+      D2 = _to_standard_form(B2,_type)
+      return true, D1^-1*D2
+   elseif _type==:symmetric
+      F = base_ring(B1)
+      n = nrows(B1)
+      A1,D1 = _block_herm_elim(B1, _type)
+      A2,D2 = _block_herm_elim(B2, _type)
+      q = order(F)
+      # eliminate all hyperbolic lines and turn A1,A2 into diagonal matrices
+      # TODO: assure that the function _elim_hyp_lines actually modifies A1 and A2
+      D1 = _elim_hyp_lines(A1)*D1
+      D2 = _elim_hyp_lines(A2)*D2
+      issquare( prod(diagonal(A1))*prod(diagonal(A2)) )[1] || return false, nothing
+      # move all the squares on the diagonal at the begin
+      _squares = [i for i in 1:n if issquare(A1[i,i])[1]]
+      our_perm = vcat(_squares, [i for i in 1:n if !(i in _squares)])      # TODO is there a more elengant way?
+      P = permutation_matrix(F,our_perm)
+      s1 = length(_squares)
+      D1 = P*D1
+      A1 = P*A1*transpose(P)
+      _squares = [i for i in 1:n if issquare(A2[i,i])[1]]
+      our_perm = vcat(_squares, [i for i in 1:n if !(i in _squares)])      # TODO is there a more elengant way?
+      P = permutation_matrix(F,our_perm)
+      s2 = length(_squares)
+      D2 = P*D2
+      A2 = P*A2*transpose(P)
+      # get same number of squares on the two diagonals of A1 and A2 by modifying A1
+      if s1!=s2
+         s = min(s1,s2)+1
+         w = A1[s,s]*A2[s,s]     # I'm sure this is not a square
+         a,b = _solve_eqn(F(1),F(1),w)
+         L = identity_matrix(F,n)
+         for i in 0:div(abs(s1-s2),2)-1
+            L[s+2*i,s+2*i] = a
+            L[s+2*i,s+2*i+1] = b*square_root(A1[s+2*i,s+2*i]*A1[s+2*i+1,s+2*i+1]^-1)
+            L[s+2*i+1,s+2*i] = b
+            L[s+2*i+1,s+2*i+1] = -a*square_root(A1[s+2*i,s+2*i]*A1[s+2*i+1,s+2*i+1]^-1)
```

Comment:
Avoid duplicating the root computation:
```suggestion
            r = square_root(A1[s+2*i,s+2*i]*A1[s+2*i+1,s+2*i+1]^-1)
            L[s+2*i,s+2*i] = a
            L[s+2*i,s+2*i+1] = b*r
            L[s+2*i+1,s+2*i] = b
            L[s+2*i+1,s+2*i+1] = -a*r
```
Actually, I'd go further:
```suggestion
            k = s+2*i
            r = square_root(A1[k,k]/A1[k+1,k+1])
            L[k:k+1,k:k+1] = [a b*r ; b -a*r]
```
(that's using the Julia way to do what your `insert_block` function does)